fullstop-web/fullstop-violations-rest/src/main/java/org/zalando/stups/fullstop/web/converter/ViolationEntityToDtoConverter.java

```diff
@@ -43,6 +43,15 @@ public Violation convert(ViolationEntity source) {
 
         violation.setUsername(source.getUsername());
 
+        if (source.getApplication() != null) {
+            violation.setApplicationId(source.getApplication().getName());
```

Comment:
`Optional.ofNullable(source.getApplication()).map(Application::getName).ifPresent(violation::setApplicationId) ` 😜 
(Just a hint how it could look like, but your solution does also work!)